common/predictive-text/worker/models/trie-model.ts

```diff
@@ -127,7 +127,7 @@
           // Delete whatever the prefix that the user wrote.
           // Note: a separate capitalization/orthography engine can take this
           // result and transform it as needed.
-          deleteLeft: leftDelOffset + prefix.length,
+          deleteLeft: leftDelOffset + prefix.kmwLength(),
```

Comment:
Does this also apply to other uses of `length`?

ll 294, 329, 362

Reply:
Line 294 will take some investigation to determine.  I'm not immediately clear on this one, but the others are safe as they are - they're not on `string`s.

Reply:
This part is what threw me off originally, then:

> Corresponds to `dn` in `com.keyman.KeyboardInterface.output`.

`com.keyman.KeyboardInterface.output` doesn't operate on code units - its use of `dn` is SMP-based.  Sorry I didn't catch it sooner - I always have to look up the difference between "code points" and "code units".

Reply:
Sounds like the docs do need updating, yes.